src/vmm/src/arch/aarch64/vcpu.rs

```diff
@@ -238,6 +257,15 @@ mod tests {
         vcpu.vcpu_init(&kvi).unwrap();
 
         setup_boot_regs(&vcpu, 0, 0x0, &mem).unwrap();
+
+        // Check that the register is reset on compatible kernels.
```

Comment:
Can we also add an integration test in `tests/integration_tests/security`?

Reply:
Integration test will be quite difficult as we will need kernel access to read the physical counter.

Reply:
There's no way to read the counter from userspace, like something like `rdmsr`?

Reply:
Actually, shouldn't we be able to access it through `KVM_GET_ONE_REG`?

Reply:
ok, let me try to implement this check through snapshots.

Reply:
Added a test with snapshot registers checking.